python_modules/libraries/dagster-aws/dagster_aws_tests/ecs_tests/launcher_tests/test_launching.py

```diff
@@ -127,6 +130,79 @@ def test_task_definition_registration(
     instance.launch_run(other_run.run_id, other_workspace)
     assert task_definitions == ecs.list_task_definitions()["taskDefinitionArns"]
 
+    # Register a new task definition if _reuse_task_definition returns False
+    # for any other reason
+    monkeypatch.setattr(instance.run_launcher, "_reuse_task_definition", lambda *_: False)
```

Comment:
I left some of the checks in here that were easy to test behaviorally (changing images, adding secrets, etc.).

But some checks would require pretty large rewrites of how our tests work (like changing IAM roles) so I pushed those into their own `test_reuse_task_definition` unit test.